packages/data/tests/unit/serializers/bard-test.js

```diff
@@ -0,0 +1,638 @@
+    id: 'metricOne',
+    metricFunctionId: undefined,
+    name: 'Metric One',
+    partialData: true,
+    source: 'dummy',
+    type: 'field',
+    valueType: 'number'
+  },
+  {
+    category: 'category',
+    id: 'metricFour',
+    metricFunctionId: 'currency|format',
+    name: 'Metric Four',
+    partialData: true,
+    source: 'dummy',
+    type: 'field',
+    valueType: 'money'
+  },
+  {
+    category: 'category',
+    id: 'metricTwo',
+    metricFunctionId: 'currency',
+    name: 'Metric Two',
+    partialData: true,
+    source: 'dummy',
+    type: 'field',
+    valueType: 'money'
+  },
+  {
+    category: 'category',
+    id: 'metricFive',
+    metricFunctionId: 'metricFunctionId take precedence over parameters',
+    name: 'Metric Five',
+    partialData: true,
+    source: 'dummy',
+    type: 'field',
+    valueType: 'number'
+  },
+  {
+    category: 'category',
+    id: 'metricThree',
+    metricFunctionId: undefined,
+    name: 'Metric Three',
+    partialData: true,
+    source: 'dummy',
+    type: 'field',
+    valueType: 'number'
+  }
+];
+
+const ParameterConvertToMetricFunction = [
+  {
+    arguments: [
+      {
+        _localValues: undefined,
+        source: 'dummy',
+        defaultValue: 'USD',
+        description: undefined,
+        expression: 'dimension:displayCurrency',
+        id: 'currency',
+        name: 'currency',
+        type: 'ref',
+        valueType: 'TEXT'
+      },
+      {
+        _localValues: undefined,
+        source: 'dummy',
+        defaultValue: 'none',
+        description: undefined,
+        expression: 'dimension:displayFormat',
+        id: 'format',
+        name: 'format',
+        type: 'ref',
+        valueType: 'TEXT'
+      }
+    ],
+    description: '',
+    id: 'currency|format',
+    name: '',
+    source: 'dummy'
+  },
+  {
+    arguments: [
+      {
+        _localValues: undefined,
+        source: 'dummy',
+        defaultValue: 'USD',
+        description: undefined,
+        expression: 'dimension:displayCurrency',
+        id: 'currency',
+        name: 'currency',
+        type: 'ref',
+        valueType: 'TEXT'
+      }
+    ],
+    description: '',
+    id: 'currency',
+    name: '',
+    source: 'dummy'
+  }
+];
+
+let Serializer;
+
+module('Unit | Serializer | metadata/bard', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    Serializer = this.owner.lookup('serializer:metadata/bard');
+  });
+
+  test('normalize `everything` with metric legacy `parameters`', function(assert) {
+    assert.deepEqual(
+      Serializer.normalize('everything', Payload),
+      {
+        metrics: Metrics,
+        dimensions: Dimensions,
+        timeDimensions: TimeDimensions,
+        tables: Tables,
+        metricFunctions: ParameterConvertToMetricFunction
+      },
+      'One metric function is created for all metrics with only the currency parameter'
+    );
+  });
+
+  test('normalize `everything` with metric functions', function(assert) {
```

Comment:
Do we have a test for when both legacy parameters and metric functions are provided? Since we're merging the constructed and provided metric functions now, it would be good to test that behavior

Reply:
https://github.com/yahoo/navi/pull/933/files#diff-7dbc9ec7e182639bf180f0b38ddae3ea

Reply:
https://github.com/yahoo/navi/blob/c468cdac82f17003357ed717d967407c4fe8af78/packages/data/tests/unit/serializers/bard-test.js#L153-L183